--- fbset-2.1/Makefile.mdkconf	2004-10-05 15:25:11.779650000 -0400
+++ fbset-2.1/Makefile	2004-10-05 15:31:46.637301827 -0400
@@ -2,7 +2,9 @@
 # Linux Frame Buffer Device Configuration
 #
 
-CC =		gcc -Wall -O2 -I.
+CC =		gcc
+CPPFLAGS =	-I.
+CFLAGS =	-O2 -Wall
 BISON =		bison -d
 FLEX =		flex
 INSTALL =	install
@@ -18,11 +20,16 @@ fbset.o:	fbset.c fbset.h fb.h
 modes.tab.o:	modes.tab.c fbset.h fb.h
 lex.yy.o:	lex.yy.c fbset.h modes.tab.h
 
-lex.yy.c:	modes.l
-		$(FLEX) modes.l
+lex.yy.c: lex.yy.h
+lex.yy.h: modes.l
+		$(FLEX) $<
+
+modes.tab.c: modes.tab.h
+modes.tab.h: modes.y
+		$(BISON) $<
 
-modes.tab.c:	modes.y
-		$(BISON) modes.y
+%.o: %.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -o $@ -c $<
 
 install:	fbset
 		if [ -f $(PREFIX)/sbin/fbset ]; then rm $(PREFIX)/sbin/fbset; fi
